app/models/agents/post_agent.rb

```diff
@@ -102,6 +102,9 @@ def handle(data, payload = {})
         when 'json'
           headers['Content-Type'] = 'application/json; charset=utf-8'
           body = data.to_json
+        when 'xml'
+          headers['Content-Type'] = 'text/xml; charset=utf-8'
+          body = data.to_xml(root: (interpolated(payload)[:xml_root] || 'post'))
```

Comment:
Do you want :xml_root from the payload, or from the Agent's options?

I'd think that we'd want this: `interpolated[:xml_root].presence || 'post'`